__tests__/app.test.js

```diff
@@ -279,4 +279,73 @@ describe("Get /api/articles/:article_id/comments", () => {
         expect(msg).toBe("Invalid input");
       })
   })
+})
+describe("POST /api/articles/:article_id/", () => {
+  test("should respond with a status 201 and the newly added comment", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "Wow, what a fantastic page",
+    };
+    return request(app)
+      .post("/api/articles/8/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body: { comment } }) => {
+        expect(comment).toEqual(
+          expect.objectContaining({
+            author: "butter_bridge",
+            body: "Wow, what a fantastic page",
+            comment_id: 19,
+            votes: 0,
+            article_id: 8,
+          })
+        )
+      })
+  })
+  test('should respond with a status 400 and "invalid input" when passed an invalid article_id', () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "Wow, what a fantastic page",
+    }
+    return request(app)
+      .post("/api/articles/ljhbsdkjhb/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Invalid input")
+      })
+  })
+  test('should respond with a status 400 and "Invalid input" when passed a comment by an invalid username ', () => {
```

Comment:
should this be 404: username does not exist?

Reply:
changed